src/main/java/byteceps/storage/Storage.java

```diff
@@ -143,6 +152,14 @@ private void loadWorkoutLogs(ExerciseManager allExercises, WorkoutManager allWor
             JSONArray exercisesArray = currentWorkout.getJSONArray(StorageStrings.EXERCISES);
             String workoutDate = currentWorkout.getString(StorageStrings.WORKOUT_DATE);
             String workoutName = currentWorkout.getString(StorageStrings.WORKOUT_NAME);
+            //validate that workoutDate is a valid date string
+            try {
+                DateTimeFormatter formatter = DateTimeFormatter.ofPattern(DayStrings.YEAR_FORMAT);
+                formatter = formatter.withLocale(formatter.getLocale());
+                LocalDate.parse(workoutDate, formatter); //ignore result, just catch exception
+            } catch (DateTimeParseException e) {
+                throw new Exceptions.InvalidInput(""); //no need for error message, LOAD_ERROR will be printed
+            }
```

Comment:
Good fix for [PE-D][Tester B] Changing the data.json to have an invalid date crashes the program #129